deepchem/utils/test/test_poly_wd_graph_utils.py

```diff
@@ -0,0 +1,80 @@
+import unittest
+from deepchem.utils import PolyWDGStringValidator
+
+
+class PolyWDGStringValidateTest(unittest.TestCase):
+    """
+    Tests for validator class to validate polymer weighted string representation
+    """
+
+    def test_get_parsed_vals(self):
+        with self.assertRaises(ValueError):
+            datapoint = "[1*]C.C[2*]|<1-2:0.5:0.5"
+            _ = PolyWDGStringValidator.get_parsed_vals(datapoint)
+
+        datapoint = "[1*]C.C[2*]|0.5|0.5|<1-2:0.5:0.5"
+        monomer_mols, fragments, polymer_rules = PolyWDGStringValidator.get_parsed_vals(
+            datapoint)
+        assert monomer_mols == "[1*]C.C[2*]"
+        assert fragments == ["0.5", "0.5"]
+        assert polymer_rules == "<1-2:0.5:0.5"
+
+    def test_get_polymer_rules(self):
+        with self.assertRaises(ValueError):
+            polymer_rules = "1-2:0.5:0.5"
+            _ = PolyWDGStringValidator.get_polymer_rules(polymer_rules)
+
+        polymer_rules = "<1-2:0.5:0.5"
+        polymer_rules = PolyWDGStringValidator.get_polymer_rules(polymer_rules)
+        assert polymer_rules == ["1-2:0.5:0.5"]
+
+    def test_valid_validate_function(self):
+        datapoint = "[1*]C.C[2*]|0.5|0.5|<1-2:0.5:0.5"
+        assert PolyWDGStringValidator().validate(datapoint)
+
+    def test_invalid_validate_function(self):
```

Comment:
there is no assert function in this function can you pls add a comment explaining the reason, if it's necessary to not have assert function.

Reply:
same comment for few more functions after this.

Reply:
It checks for invalid format, so there is assertRaise. It tests that the given code returns expected errors when a certain kind of invalid format is provided.

Reply:
For each invalid input format there is no assert rather assert raises.

Reply:
```python
try:
    .....
except exception as e:
    .....
    assert e is ValueError
```
The above code block is equivalent to
```python
with self.assertRaise(valueError):
    ....
```
It just makes the code short and concise (inside the unittest)
Hence I have used it to validate the error raise.